apps/server/test/lexical/server/code_intelligence/completion/translations/function_test.exs

```diff
@@ -200,5 +200,27 @@ defmodule Lexical.Server.CodeIntelligence.Completion.Translations.FunctionTest d
 
       assert completion.sort_text == "info/1"
     end
+
+    test "dunder functions have lower completion priority", %{project: project} do
+      completions =
+        complete(project, "GenServer.|")
+        |> Enum.sort_by(& &1.sort_text)
+
+      total_completions = length(completions)
+      info_idx = Enum.find_index(completions, &(&1.sort_text == "info/1"))
+
+      assert info_idx > total_completions * 0.80
```

Comment:
I did some work on this:

```ex
    test "dunder functions have lower completion priority", %{project: project} do
      completions =
        project
        |> complete("GenServer.|")
        |> Enum.sort_by(& &1.sort_text)
        |> Enum.with_index()

      is_dunder = fn {fun, _} -> String.starts_with?(fun.label, "__") end
      dunder_functions = Enum.filter(completions, is_dunder)
      non_dunder = Enum.reject(completions, is_dunder)

      for {_f, idx} <- dunder_functions do
        assert Enum.all?(non_dunder, &smaller_index(&1, idx))
      end
    end

    test "default functions have lower completion priority", %{project: project} do
      completions =
        project
        |> complete("GenServer.|")
        |> Enum.sort_by(& &1.sort_text)
        |> Enum.with_index()

      is_default = fn {fun, _} -> fun.sort_text in ["module_info/0"] end
      default_funs = Enum.filter(completions, is_default)
      other_funs = Enum.reject(completions, is_default)

      for {_f, idx} <- default_funs do
        assert Enum.all?(other_funs, &smaller_index(&1, idx))
      end
    end
  end

  defp smaller_index({_, i}, idx), do: i < idx
```

But I don't like it very much, using this logic, the dunder case fails because the `module_info` function is not a dunder function and gets after it, and I don't think it would be ideal to filter it out in the dunder test case.

Maybe a simpler solution would be: get all the dunder and default functions and assert they do not start with a booster character, what do you think @scohen?

Reply:
> Maybe a simpler solution would be: get all the dunder and default functions and assert they do not start with a booster character, what do you think

it might even be easier than that. Make a function called `boosted?` that checks for a boost character. Then it's 
```elixir
completions = 
project
|> complete("GenServer.|")
|> Enum.sort_by(& &1.sort_text)

{dunder_completions, non_dunder_completions} = Enum.split_with(completions, &String.starts_with?(&1.label, "__"))

for completion <- dunder_completions do 
  refute boosted?(completion)
end

for completion <- non_dunder_functions do 
  assert boosted?(completion)
end
```